### PR TITLE
take time format and tick interval

### PR DESCRIPTION
**Pull request by atago0129**
* Roadmappy is taken `timeFormat` and `tickInterval`
* bug fix on https://github.com/atago0129/roadmappy/pull/9

### CODE REVIEW

File: src/component/RoadmapCanvas.js [hunk 1]
```diff
@@ -8,14 +8,16 @@ export class RoadmapCanvas {
...
+    timeFormat: d3.timeFormat("%b %d"),
+    tickInterval: d3.timeMonday
   };
 
   /**
    * @param {object} options
    */
   constructor(options) {
-    this.type = AbstractRoadmapGroup.isValidType(type) ? options.type : AbstractRoadmapGroup.TYPE.STORY;
+    this.type = AbstractRoadmapGroup.isValidType(options.type) ? options.type : AbstractRoadmapGroup.TYPE.STORY;
```
Comment (atago0129) [inline src/component/RoadmapCanvas.js]: bug fix

File: src/component/RoadmapCanvas.js [hunk 2]
```diff
@@ -28,6 +30,12 @@ export class RoadmapCanvas {
     if (style.barHeight !== undefined) {
       this.style.barHeight = parseInt(style.barHeight);
     }
+    if (style.timeFormat !== undefined) {
```
Comment (atago0129) [inline src/component/RoadmapCanvas.js]: Should I validate `timeFormat` ...?

File: debug/index.html [hunk 3]
```diff
@@ -8,7 +8,12 @@
       var roadmappy = new window.Roadmappy.default(
         {
           target: "roadmappy",
-          type: "all"
+          type: "all",
```
Comment (hrsh7th) [inline debug/index.html:11]: type: all was removed.

File: debug/index.html [hunk 4]
```diff
@@ -8,7 +8,12 @@
       var roadmappy = new window.Roadmappy.default(
         {
           target: "roadmappy",
-          type: "all"
+          type: "all",
+          style: {
+            "barHeight": 17,
+            "timeFormat": "%-m月%-d日", // required d3 time format
+            "tickInterval": "timeMonday" // required d3 interval format
+          }
```
Comment (hrsh7th) [inline debug/index.html:16]: nice interface.

File: src/component/RoadmapCanvas.js (hunk 1 above)
Comment (hrsh7th) [reply src/component/RoadmapCanvas.js]: sorry and thx.

File: src/component/RoadmapCanvas.js (hunk 2 above)
Comment (hrsh7th) [reply src/component/RoadmapCanvas.js]: hum... It's is not wrong. But bit ugly...

I think, We should create `RoadmapOption`.
What dot you think?

File: debug/index.html (hunk 3 above)
Comment (atago0129) [reply debug/index.html:11]: oh, yes

File: src/component/RoadmapCanvas.js (hunk 2 above)
Comment (atago0129) [reply src/component/RoadmapCanvas.js]: Nice idea!
I will try it!

**Pull request merged**